### PR TITLE
[ES6 class] Update plugin store es6

### PR DESCRIPTION
**Pull request by mlunoe**
---
~~⚠️ This PR will fail until @orlandohohmeier's PR has merged~~
~~⚠️ Is dependent on this PR https://github.com/dcos/mesosphere-shared-reactjs/pull/4~~
⚠️ All plugins that depends on this needs to update their stores to use `addStoreConfig` in stead of `createPluginStore`

---

This PR is removing the concept of `createPluginStore` and replacing it with `addStoreConfig` to be used in the plugin stores.

To test this out `git checkout mlunoe/plugin-store-es6-update-test`

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @pierlo-upitup

**Comment by pierlo-upitup** [comment]
LGTM

**Comment by kennyt** [comment]
can't merge because integration failing

File: src/js/stores/__tests__/BaseStore-test.js [hunk 1]
```diff
@@ -1,63 +1,37 @@
-jest.dontMock('../BaseStore');
-
 var BaseStore = require('../BaseStore');
 
 describe('BaseStore', function () {
 
   beforeEach(function () {
     this.instance = new BaseStore();
+    this.instance.on = jasmine.createSpy('eventOn');
```
Comment (mlunoe) [inline src/js/stores/__tests__/BaseStore-test.js]: These are new tests. The old one test are moved to GetSetBaseStore-test.js and are unchanged.

File: src/js/stores/__tests__/GetSetBaseStore-test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+var GetSetBaseStore = require('../GetSetBaseStore');
```
Comment (mlunoe) [inline src/js/stores/__tests__/GetSetBaseStore-test.js]: These tests are unchanged and moved from BaseStore-test.js

**Pull request merged**